File: 2023/Leetcode/problem121.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let profit = 0;
    let buyPrice = prices[0];

    for (let i = 1; i < prices.length; i++) {
        if (buyPrice > prices[i]) {
            buyPrice = prices[i];
        } 

        if (prices[i] - buyPrice > profit) {
            profit = prices[i] - buyPrice;
        }
    }

    return profit;
};

File: 2023/Leetcode/problem121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buyPrice = prices[0]
        profit = 0
        for i in range(1, len(prices)):
            if prices[i] < buyPrice:
                buyPrice = prices[i]
            if prices[i] - buyPrice > profit:
                profit = prices[i] - buyPrice

        return profit
                